lib/bioinfo/deeptools.ml

```diff
@@ -0,0 +1,190 @@
+      option (flag string "--centerReads") centerreads ;
+      option (opt "--samFlagInclude" int) samflaginclude ;
+      option (opt "--samFlagExclude" int) samflagexclude ;
+      option (opt "--minFragmentLength" int) minfragmentlength ;
+      option (opt "--maxfragmentLength" int) maxfragmentlength ;
+    ]
+      other_args
+  )
+
+
+let bamcoverage ?outfileformat ?scalefactor ?filterrnastrand ?binsize ?blacklistfilename
+    ?(threads = 1) ?normalizeto1x ?normalizeusingrpkm ?ignorefornormalization
+    ?skipnoncoveredregions ?smoothlength ?extendreads ?ignoreduplicates
+    ?minmappingquality ?centerreads ?samflaginclude ?samflagexclude
+    ?minfragmentlength ?maxfragmentlength bam =
+  workflow ~descr:"bamcoverage" ~np:threads ~mem:(3 * 1024) [
+    bam_gen_cmd "bamCoverage" [
+      option (opt "--filterRNAstrand" filterRNAstrand_expr) filterrnastrand ;
+      option (opt "--binSize" int) binsize ;
+      opt "--numberOfProcessors" ident np ;
+      string "--bam" ; (dep bam) // "reads.bam"  ;
+      opt "--outFileName" ident dest ;
+    ]
+  ]
+
+let bamcompare ?outfileformat ?scalefactormethod ?samplelength ?numberofsamples
+    ?scalefactor ?ratio ?pseudocount ?binsize ?region ?blacklistfilename ?(threads = 1)
+    ?normalizeto1x ?normalizeusingrpkm ?ignorefornormalization ?skipnoncoveredregions
+    ?smoothlength ?extendreads ?ignoreduplicates ?minmappingquality
+    ?centerreads ?samflaginclude ?samflagexclude ?minfragmentlength
+    ?maxfragmentlength bam1 bam2 =
+  workflow ~descr:"bamcompare" ~np:threads ~mem:(3 * 1024) [
+    bam_gen_cmd "bamCompare" [
+      option (opt "--scaleFactorMethod" scalefactormethod_expr) scalefactormethod ;
+      option (opt "--sampleLength" int) samplelength ;
+      option (opt "--numberOfSamples" int) numberofsamples ;
+      option (opt "--ratio" ratio_expr) ratio ;
+      option (opt "--pseudocount" int) pseudocount ;
+      option (opt "--binSize" int) binsize ;
+      option (opt "--region" string) region ;
+      opt "--numberOfProcessors" ident np ;
+      string "--bamfile1" ; (dep bam1) // "reads.bam"  ;
+      string "--bamfile2" ; (dep bam2) // "reads.bam"  ;
+      opt "--outFileName" ident dest ;
+    ]
+  ]
+
+
+let bigwigcompare ?outfileformat ?scalefactor ?ratio ?pseudocount ?binsize
+    ?region ?blacklistfilename ?(threads = 1)
+     bigwig1 bigwig2 =
+  workflow ~descr:"bigwigcompare" ~np:threads ~mem:(3 * 1024) [
+    cmd "bigwigCompare" ~env [
+      option (opt "--outFileFormat" file_format_expr) outfileformat ;
+      option (opt "--scaleFactor" string) scalefactor ;
+      option (opt "--ratio" ratio_expr) ratio ;
+      option (opt "--pseudocount" int) pseudocount ;
+      option (opt "--binSize" int) binsize ;
+      option (opt "--region" string) region ;
+      option (opt "--blackListFileName" dep) blacklistfilename ;
+      opt "--numberOfProcessors" ident np ;
+      string "--bigwig1" ; (dep bigwig1) // "reads.bw"  ;
```

Comment:
why not have `opt "--bigwig1" dep bigwig1` instead? It seems unncessary complicated to ask the bigWig to be in a directory...